src/Compilers/CSharp/Portable/Symbols/TypeParameterSymbol.cs

```diff
@@ -454,9 +444,43 @@ internal bool IsReferenceTypeFromConstraintTypes(ImmutableArray<TypeSymbolWithAn
             return AnyConstraintTypes(constraintTypes, inProgress, (type, arg) => ConstraintImpliesReferenceType(type.TypeSymbol, arg));
         }
 
-        internal bool IsNotNullableIfReferenceTypeFromConstraintTypes(ImmutableArray<TypeSymbolWithAnnotations> constraintTypes, ConsList<TypeParameterSymbol> inProgress)
+        internal bool? IsNotNullableIfReferenceTypeFromConstraintTypes(ImmutableArray<TypeSymbolWithAnnotations> constraintTypes, ConsList<TypeParameterSymbol> inProgress)
```

Comment:
>IsNotNullableIfReferenceTypeFromConstraintTypes [](start = 23, length = 47)

nit: It feels like this would be easier to follow if it was inverted: "IsNullableIf...". The `bool?` would have the same meaning as in other places that deal with nullability.

Reply:
> It feels like this would be easier to follow if it was inverted: "IsNullableIf...". The bool?

I think that will lead to a confusion. If result of this function is false, it doesn't mean that the type parameter represents nullable type, it can represent either.

---
In reply to: [213847892](https://github.com/dotnet/roslyn/pull/29586#discussion_r213847892) [](ancestors = 213847892)